smartdashboard/views.py

```diff
@@ -341,3 +345,467 @@ def __init__(
+
+    def _update_table(self, dframe: pd.DataFrame) -> None:
+        """Update client table for selected shard
+
+        :param dframe: DataFrame with client data
+        :type dframe: pandas.DataFrame
+        """
+
+        def process_dataframe(dframe: pd.DataFrame) -> pd.DataFrame:
+            dframe = dframe.loc[dframe["timestamp"] == dframe["timestamp"].max()]
+            dframe = dframe.drop(columns=["timestamp"])
+            dframe = dframe.sort_values(by="client_id")
+            dframe = dframe.rename(
+                columns={
+                    "client_id": "Client ID",
+                    "address": "Address",
+                }
+            )
+            return dframe
+
+        self.table_element.dataframe(
+            process_dataframe(dframe), use_container_width=True, hide_index=True
+        )
+
+    def _update_graph(self, dframe: pd.DataFrame) -> None:
+        """Update client graph for selected shard
+
+        :param dframe: DataFrame with client data
+        :type dframe: pandas.DataFrame
+        """
+
+        def process_dataframe(dframe: pd.DataFrame) -> pd.DataFrame:
+            dframe["timestamp"] = (dframe["timestamp"] - self.timestamp_min) / 1000
+            return dframe
+
+        if self.chart is None or self.sampling:
+            chart = (
+                alt.Chart(process_dataframe(dframe))
+                .mark_line()
+                .encode(
+                    x=alt.X("timestamp:Q", axis=alt.Axis(title="Timestep in seconds")),
+                    y=alt.Y("num_clients:Q", axis=alt.Axis(title="Client Count")),
+                    tooltip=["timestamp:Q", "num_clients:Q"],
+                )
+                .interactive()
+                .properties(
+                    height=500,
+                    title=alt.TitleParams("Total Client Count", anchor="middle"),
+                )
+            )
+
+            self.graph_element.altair_chart(chart, use_container_width=True)
+            self.chart = chart
+
+        else:
+            self.graph_element.add_rows(process_dataframe(dframe))
+
+
+class OrchestratorSummaryView(ViewBase):
+    """View class for orchestrator summary section of the Database Telemetry page"""
+
+    def __init__(self, orchestrator: t.Optional[Orchestrator]) -> None:
+        """Initialize an OrchestratorSummaryView
+
+        :param orchestrator: Selected orchestrator
+        :type orchestrator: t.Optional[Orchestrator]
+        """
+        self.orchestrator = orchestrator
+        self.status_element = DeltaGenerator()
+
+    @property
+    def status(self) -> str:
+        """Get orchestrator status summary
+
+        :return: Status summary
+        :rtype: str
+        """
+        return get_orchestrator_status_summary(self.orchestrator)
+
+    def update(self) -> None:
+        """Update status element in OrchestratorView"""
+        self.status_element.write(self.status)
+
+
+class TelemetryView(ViewBase):
```

Comment:
Pedantic naming thing, just in case we decide to add more telemetry for different types of entities in future:
```suggestion
class DatabaseTelemetryView(ViewBase):
```

Reply:
fixed!